src/Enhavo/Bundle/AppBundle/Type/TypeCollector.php

```diff
@@ -18,44 +19,36 @@ class TypeCollector implements CollectorInterface
      */
     private $collection;
 
+    private $container;
+
     /**
      * @var string
      */
     private $typeName;
 
-    public function __construct($typeName = 'Type')
+    public function __construct(ContainerInterface $container, $typeName = 'Type')
     {
         $this->collection = array();
+        $this->container = $container;
         $this->typeName = $typeName;
     }
 
-    public function add(TypeInterface $type)
+    public function add($alias, $id)
     {
-        $this->collection[$type->getType()] = $type;
-    }
-
-    /**
-     * @return TypeInterface[]
-     */
-    public function getCollection()
-    {
-        $collection = [];
-        foreach($this->collection as $type) {
-            $collection[] = $type;
-        }
-        return $collection;
+        $this->collection[$alias] = $id;
     }
 
-    public function getType($name)
+    public function getType($alias)
     {
-        if(isset($this->collection[$name])) {
-            return $this->collection[$name];
+        if(isset($this->collection[$alias])) {
+            $serviceId = $this->collection[$alias];
+            return $this->container->get($serviceId);
```

Comment:
Please check if TypeInterface is used and ``getType``will return the alias as well. If not, just throw an own Exception e.g. ``TypeNotValidException``